### PR TITLE
Make Checkstyle integration test macOS-compatible

### PR DESCRIPTION
**Pull request by rickie**

### CODE REVIEW

**Comment by rickie** [comment]
Either @benhalasi or @mohamedsamehsalah can you try to run this on your Mac and let us know if it works?

**Comment by rickie** [comment]
Ah, I mean the `./integration-tests/checkstyle-10.12.4.sh` from the root folder. 😄

**Comment by benhalasi** [comment]
I've run it:
```
> ./integration-tests/checkstyle-10.12.4.sh                                       
usage: paste [-s] [-d delimiters] file ...
xargs: ggrep: terminated with signal 13; aborting
```

**Comment by rickie** [comment]
Thanks @benhalasi , I added something I didn't move over. Could you try again?

**Comment by benhalasi** [comment]
Yep, it runs now, but it says there are unexpected changes:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (error-prone-compile) on project checkstyle: Fatal error compiling: DeleteEmptyMethod is not a valid checker name -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Inspecting changes...
--- /Users/benhalasi/PicnicSupermarket/error-prone-support/integration-tests/checkstyle-10.12.4-expected-changes.patch  2023-11-11 08:39:51
+++ /var/folders/2b/q73x_kyx6s10rrdnjwtv2zb80000gn/T/tmp.C4fad86ddk/checkstyle-10.12.4-changes.patch    2023-11-13 09:33:21
@@ -51469,17 +51469,17 @@
 -import java.util.stream.Collectors;
  import javax.xml.parsers.ParserConfigurationException;
  import org.junit.jupiter.api.BeforeEach;
- import org.junit.jupiter.api.Test;
-@@ -71,7 +70,7 @@ import org.w3c.dom.NamedNodeMap;
- import org.w3c.dom.Node;
+ import org.junit.jupiter.api.Disabled;
+@@ -73,7 +72,7 @@ import org.w3c.dom.Node;
  import org.w3c.dom.NodeList;
  
+ @Disabled("Reason: https://github.com/junit-team/junit5/issues/3536.")
 -public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
 +final class XdocsJavaDocsTest extends AbstractModuleTestSupport {
    private static final Map<String, Class<?>> FULLY_QUALIFIED_CLASS_NAMES =
        ImmutableMap.<String, Class<?>>builder()
            .put("int", int.class)
-@@ -103,16 +102,15 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -105,16 +104,15 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
            .build();
  
    private static final Set<String> NON_BASE_TOKEN_PROPERTIES =
@@ -51505,7 +51505,7 @@
  
    private static final List<List<Node>> CHECK_PROPERTIES = new ArrayList<>();
    private static final Map<String, String> CHECK_PROPERTY_DOC = new HashMap<>();
-@@ -130,7 +128,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -132,7 +130,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
    }
  
    @BeforeEach
@@ -51514,7 +51514,7 @@
      final DefaultConfiguration checkConfig =
          new DefaultConfiguration(JavaDocCapture.class.getName());
      checker = createChecker(checkConfig);
-@@ -144,7 +142,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -146,7 +144,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
     *     method
     */
    @Test
@@ -52239,9 +52239,9 @@
  import java.util.stream.Stream;
  import org.itsallcode.io.Capturable;
  import org.itsallcode.junit.sysextensions.SystemOutGuard;
-@@ -40,10 +41,10 @@ import org.junit.jupiter.api.Test;
- import org.junit.jupiter.api.extension.ExtendWith;
+@@ -42,10 +43,10 @@ import org.junit.jupiter.api.extension.ExtendWith;
  
+ @Disabled("Reason: https://github.com/junit-team/junit5/issues/3536.")
  @ExtendWith(SystemOutGuard.class)
 -public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
 +final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
@@ -52252,7 +52252,7 @@
            "Checker",
            "TreeWalker",
            "JavadocMetadataScraper",
-@@ -67,7 +68,7 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
+@@ -69,7 +70,7 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
     *     System.out} for error messages
     */
    @Test
@@ -52261,7 +52261,7 @@
        throws Exception {
      systemOut.captureMuted();
  
-@@ -117,12 +118,12 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
+@@ -119,12 +120,12 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
                .filter(path -> !path.toString().endsWith(".properties"))
                .map(MetadataGeneratorUtilTest::getMetaFileName)
                .sorted()
There are unexpected changes.

```

**Comment by rickie** [comment]
Ah those were the changes that I introduced, they shouldn't be there. Let me fix that. 

Thanks for checking @benhalasi 🚀 !

File: integration-tests/checkstyle-10.12.4-init.patch [hunk 1]
```diff
@@ -1,3 +1,5 @@
+diff --git a/pom.xml b/pom.xml
+index ca1f3d5..33ad480 100644
```
Comment (Stephan202) [inline integration-tests/checkstyle-10.12.4-init.patch]: Here and below: let's omit this meta-data.

**Review by Stephan202 — COMMENTED** [review]
Rebased and added a commit; further review TBD.

**Comment by rickie** [comment]
@benhalasi Stephan helped me out and made the branch correct. I'd say that right now you should be able to have a successful run. Can you check again 😬 ?

**Review by rickie — COMMENTED** [review]
Oh I see I had some pending comments.

File: integration-tests/checkstyle-10.12.4-init.patch [hunk 2]
```diff
@@ -106,6 +114,8 @@
          final String rendered = report
              .replace(VERSION_PLACEHOLDER, String.valueOf(version))
              .replace(RESULTS_PLACEHOLDER, String.join(",\n", results));
+diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java b/src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java
```
[the remark this answers is not in the document]
Comment (rickie) [reply integration-tests/checkstyle-10.12.4-init.patch]: Lol, you now fixed this @Stephan202 :). Thanks.

**Comment by benhalasi** [comment]
I've run it on `[2023-11-19] [0e9b1835] | Minimize initial patch {{Stephan Schroevers}}  (HEAD -> rossendrijver/mac_compatibility, origin/rossendrijver/mac_compatibility)`:
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.100 s
[INFO] Finished at: 2023-11-20T09:01:49+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (error-prone-compile) on project checkstyle: Fatal error compiling: DeleteEmptyMethod is not a valid checker name -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Inspecting changes...
--- /Users/benhalasi/PicnicSupermarket/error-prone-support/integration-tests/checkstyle-10.12.4-expected-changes.patch  2023-11-11 08:39:51
+++ /var/folders/2b/q73x_kyx6s10rrdnjwtv2zb80000gn/T/tmp.1pGG6DoQLc/checkstyle-10.12.4-changes.patch    2023-11-20 09:01:50
@@ -51469,17 +51469,17 @@
 -import java.util.stream.Collectors;
  import javax.xml.parsers.ParserConfigurationException;
  import org.junit.jupiter.api.BeforeEach;
- import org.junit.jupiter.api.Test;
-@@ -71,7 +70,7 @@ import org.w3c.dom.NamedNodeMap;
- import org.w3c.dom.Node;
+ import org.junit.jupiter.api.Disabled;
+@@ -73,7 +72,7 @@ import org.w3c.dom.Node;
  import org.w3c.dom.NodeList;
  
+ @Disabled("Reason: https://github.com/junit-team/junit5/issues/3536.")
 -public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
 +final class XdocsJavaDocsTest extends AbstractModuleTestSupport {
    private static final Map<String, Class<?>> FULLY_QUALIFIED_CLASS_NAMES =
        ImmutableMap.<String, Class<?>>builder()
            .put("int", int.class)
-@@ -103,16 +102,15 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -105,16 +104,15 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
            .build();
  
    private static final Set<String> NON_BASE_TOKEN_PROPERTIES =
@@ -51505,7 +51505,7 @@
  
    private static final List<List<Node>> CHECK_PROPERTIES = new ArrayList<>();
    private static final Map<String, String> CHECK_PROPERTY_DOC = new HashMap<>();
-@@ -130,7 +128,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -132,7 +130,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
    }
  
    @BeforeEach
@@ -51514,7 +51514,7 @@
      final DefaultConfiguration checkConfig =
          new DefaultConfiguration(JavaDocCapture.class.getName());
      checker = createChecker(checkConfig);
-@@ -144,7 +142,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
+@@ -146,7 +144,7 @@ public class XdocsJavaDocsTest extends AbstractModuleTestSupport {
     *     method
     */
    @Test
@@ -52239,9 +52239,9 @@
  import java.util.stream.Stream;
  import org.itsallcode.io.Capturable;
  import org.itsallcode.junit.sysextensions.SystemOutGuard;
-@@ -40,10 +41,10 @@ import org.junit.jupiter.api.Test;
- import org.junit.jupiter.api.extension.ExtendWith;
+@@ -42,10 +43,10 @@ import org.junit.jupiter.api.extension.ExtendWith;
  
+ @Disabled("Reason: https://github.com/junit-team/junit5/issues/3536.")
  @ExtendWith(SystemOutGuard.class)
 -public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
 +final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
@@ -52252,7 +52252,7 @@
            "Checker",
            "TreeWalker",
            "JavadocMetadataScraper",
-@@ -67,7 +68,7 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
+@@ -69,7 +70,7 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
     *     System.out} for error messages
     */
    @Test
@@ -52261,7 +52261,7 @@
        throws Exception {
      systemOut.captureMuted();
  
-@@ -117,12 +118,12 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
+@@ -119,12 +120,12 @@ public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
                .filter(path -> !path.toString().endsWith(".properties"))
                .map(MetadataGeneratorUtilTest::getMetaFileName)
                .sorted()
There are unexpected changes.
```

**Comment by rickie** [comment]
Rebased and resolved conflict.

**Comment by rickie** [comment]
@benhalasi or @mohamedsamehsalah can one of you take a look and try to run the `./integration-tests/checkstyle.sh`? 😄

**Review by benhalasi — APPROVED** [review]
🚀

run on MacOS, openjdk 17.0.3 2022-04-19
```
mvnd -DskipTests -Dverification.skip clean install 
./integration-tests/checkstyle.sh
git status
```

didn't produce any diff!

**Review by Stephan202 — APPROVED** [review]
I rebased the branch and added two commits.

Suggested commit message:
```
Make Checkstyle integration test macOS-compatible (#867)
```

NB: Initially I observed the following error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (error-prone-compile) on project checkstyle: Fatal error compiling: DeleteEmptyMethod is not a valid checker name -> [Help 1]
```

This happened because the `workshop/target` directory of the `workshop` Git branch still existed. This is a bug in the script we may at some point want to fix. (But it's not related to the changes in this PR.)

**Comment by Stephan202** [comment]
> /integration-test

This build was [skipped](https://github.com/PicnicSupermarket/error-prone-support/actions/runs/7769273954?pr=867+%28comment%29). Not sure why :thinking:.

**Comment by rickie** [comment]
It ran successfully when I posted a comment 🤔. So I'd say let's merge, right? Rebased.

**Comment by rickie** [comment]
See: https://github.com/PicnicSupermarket/error-prone-support/actions/runs/7845898450?pr=867+%28comment%29.

**Comment by Stephan202** [comment]
Sure, then let's merge :+1:. (I would also have expected it to show up in the check overview, but alas.)

**Pull request merged**